src/lib/crypto/kimchi_backend/gadgets/keccak.ml

```diff
@@ -552,34 +560,86 @@ let hash (type f)
         pad_101 (module Circuit) message rate
   in
   let hash = sponge (module Circuit) padded ~length ~capacity ~rate in
-  check_bytes (module Circuit) hash ;
+  (* Check each cvar output is 8 bits at most *)
+  if byte_checks then check_bytes (module Circuit) hash ;
```

Comment:
The usual model for constraints is that inputs should not be constrained (we can assume that whatever created them also constrained them), but outputs should be.

So, I think the hash should be passed to `check_bytes` regardless of `byte_checks`, because this function creates and returns it

Reply:
Oh, okay your point makes sense, let me adapt this in the next commit